File: Exámenes/2020/Septiembre/haskell/BinPackingDemo.hs
{------------------------------------------------------------------------------
 - Data Structures, Grado en Informática. UMA.
 -
 - First Fit algorithm for Bin Packing problem
 -----------------------------------------------------------------------------}
 
import AVL

list1 :: [Weight]
list1 = [9, 9, 9, 9, 9, 9, 9, 9, 9]

list2 :: [Weight]
list2 = [5, 7, 5, 2, 4, 2, 5, 1, 6]

solve :: Capacity -> [Weight] -> [Bin]
solve w os = toList (addAll w os)

solveLinear :: Capacity -> [Weight] -> [Bin]
solveLinear w oo = seqToList (linearBinPacking w oo) 

main = do
  putStrLn ("Solving W = 10 and "++show list1)
  mapM_ print (solve 10 list1)
  putStrLn ("\n--------------------\n")  
  putStrLn ("Solving W = 10 and "++show list2)
  mapM_ print (solve 10 list2)
  
  {-
  -- Para alumnos sin evaluacion continua
  -- ONLY FOR STUDENTS WITHOUT CONTINUOUS ASSESSMENT

  
  putStrLn("LinearBinPacking W=10 and "++show list1)
  mapM_ print (solveLinear 10 list1)
  putStrLn ("\n--------------------\n")  
  putStrLn ("LinearBinPacking W=10 and "++show list2)
  mapM_ print (solveLinear 10 list2)
  -}

File: Exámenes/2020/Septiembre/Java/java/src/BinPackingDemo.java
/**
 * @author Pepe Gallardo, Data Structures, Grado en Informática. UMA.
 *
 * Tests for First Fit algorithms for Bin Packing problem
 */

import dataStructures.list.List;

public class BinPackingDemo {
    public static void main(String[] args) {
        main1(args);
        System.out.println();
        main2(args);
    }

    public static void test(int initialCapacity, int[] weights) {
        System.out.printf("Solving a problem instance with %d as initial capacity and these weights", initialCapacity);
        for(int weight : weights)
            System.out.printf(" %d",weight);
        System.out.println(".");
        AVL avl = new AVL();
        avl.addAll(initialCapacity, weights);
        List<Bin> bins = avl.toList();
        System.out.println("Solution is: ");
        System.out.println(bins);
    }

    public static void main1(String[] args) {
        test(10, new int[]{ 9, 9, 9, 9, 9, 9, 9, 9, 9 });
    }

    public static void main2(String[] args) {
        test(10, new int[]{ 5, 7, 5, 2, 4, 2, 5, 1, 6 });
    }
}